statgen/pheweb
# Implement "show correlated phenotypes" feature on pheno page

# Purpose
Per @sgagliano  / @dtaliun request, implements a "show correlated phenotypes" table on the phenotype summary page, under the manhattan/ QQ plot.

<img width="1417" alt="screen shot 2019-03-05 at 4 47 20 pm" src="https://user-images.githubusercontent.com/2957073/53839966-dd170c00-3f66-11e9-84dc-22e33f925850.png">

## Design notes
The design spec calls for this to be an optional feature. The basic ingest step (`pheweb process`) will warn if the correlation file is missing, but only fail if the feature has been explicitly requested (via the config file). This new piece of data can be incrementally added to an existing pheweb by running a single new command. (see below)

The rg pipeline file format is still being worked out, and the current iteration is not very tolerant of file format inconsistencies. Eventually we may want to incorporate the pipeline directly into Pheweb, to ensure more consistent data processing.

# How to test this
(TODO: add test data!)

The below instructions should reflect what has been added to README.md.

The correlation file can be incorporated by copying `pheno-correlations.txt` into the same folder as `pheno-list.json`, and then running the new subcommand:

`$ phweb pheno-correlation`

To see the table, make sure that your config.py file specifies `show_correlations = True`, then run pheweb serve. The demonstration data provided is intended to show correlations for the snowstorm phenotype: `http://0.0.0.0:5000/pheno/snowstorm`

# TODO
- [x] Add unit tests and better sample data to help generate appropriate plots.
- [x] Each table row should link to the page for the correlated phenotype
- Future: incorporate the [RG pipeline](https://github.com/statgen/pheweb-rg-pipeline) as a direct pheweb data processing step, rather than requiring the user to import an externally generated file.
